### ISSUE TITLE
[clang-tidy]llvm-header-guard false positive on #pragma once header

### ISSUE THREAD

**Issue by mzyKi** [llvm/llvm-project#111475]
a.hpp
```cpp
#pragma once
int i = 0;
```

test.cpp
```
#include "a.hpp"
```
Result
```bash
$ clang-tidy -checks=-*,llvm-header-guard test.cpp -header-filter=.*
a.hpp:1:1: warning: header is missing header guard [llvm-header-guard]
#pragma once
^
```

I think this check may give one inaccurate note here. Is it necessary to make changes to this situation? I want some suggestion about revising ```llvm-header-guard```.  Do not report this bug or give another note?
